src/main/scala/net/snowflake/spark/snowflake/Parameters.scala

```diff
@@ -196,11 +202,29 @@ object Parameters {
         "A snowflake user must be provided with '" + PARAM_SF_USER + "' parameter, e.g. 'user1'"
       )
     }
+    val authenticatorVal = userParameters.getOrElse(PARAM_AUTHENTICATOR, "")
     if ((!userParameters.contains(PARAM_SF_PASSWORD)) &&
-        (!userParameters.contains(PARAM_PEM_PRIVATE_KEY))) {
+        (!userParameters.contains(PARAM_PEM_PRIVATE_KEY)) &&
+        // 2020-02-13 - if OAuth token not provided
+        (!authenticatorVal.equals("oauth"))) {
       throw new IllegalArgumentException(
         "A snowflake password or private key path must be provided with '" + PARAM_SF_PASSWORD +
-          " or " + PARAM_PEM_PRIVATE_KEY + "' parameter, e.g. 'password'"
+          " or " + PARAM_PEM_PRIVATE_KEY + " or " + PARAM_OAUTH_TOKEN + "' parameter, e.g. 'password'"
+      )
+    }
+    // 2020-02-13 - ensure OAuth token  provided if PARAM_AUTHENTICATOR = OAuth
+    val tokenVal = userParameters.getOrElse(PARAM_OAUTH_TOKEN, "")
```

Comment:
use 'sfToken()'

Reply:
confirmed not possible, the sfToken is not callable from the mergeParameters method

Reply:
I also added JIRA subtask with functionality / usage (please see SNOW-140395)